test/parallel/test-assert.js

```diff
@@ -453,8 +453,37 @@ assert.throws(makeBlock(thrower, TypeError));
                      'a.doesNotThrow is not catching type matching errors');
 }
 
-assert.throws(function() { assert.ifError(new Error('test error')); },
-              /^Error: test error$/);
+
+// Assert that the exception is wraped with an AssertionError
+assert.throws(
+  function() { assert.ifError(new Error('test error')); },
+  common.expectsError({
+                code: 'ERR_ASSERTION',
+                type: a.AssertionError,
+                message: /^ifError failed\. Original error:$/m
+  })
+);
+// Assert that the original exception is inside
+assert.throws(
+  function() { assert.ifError(new Error('test_error_slug')); },
+  common.expectsError({
+                code: 'ERR_ASSERTION',
+                type: a.AssertionError,
+                message: /^Error: test_error_slug$/m
```

Comment:
I am not sure how this is any different from the test above. I actually would expect this test to fail because I would have expected it to be wrapped in the `ifError failed` part.